api/v1/submissions/PKPSubmissionController.php

```diff
@@ -1706,4 +1758,41 @@ protected function getWriteDisabledErrors(string $schemaName, array $params): ar
 
         return $errors;
     }
+
+    /**
+     * Copy author, files, etc. multilingual fields from old to new changed language
+     */
+    protected function copyMultilingualData(Submission $submission, string $newLocale): void {
+        $oldLocale = $submission->getData('locale');
+        $editProps = fn (Author|SubmissionFile $item, array $props): array => collect($props)
+            ->mapWithKeys(fn (string $p): array => [$p => ($d = $item->getData($p)[$oldLocale] ?? null) ? [$newLocale => $d] : null])
+            ->filter()
+            ->toArray();
+
+        // Submission files
+        $fileProps = [
+            'name',
+        ];
+        Repo::submissionFile()
+            ->getCollector()
+            ->filterBySubmissionIds([$submission->getId()])
+            ->getMany()
+            ->each(fn (SubmissionFile $f) => Repo::submissionFile()->edit($f, $editProps($f, $fileProps)));
+        
+        // Contributor
+        $contributorProps = [
```

Comment:
There are other localized fields such as bio and affiliation. Just wanted to flag that to make sure those shouldn't be copied over as well.

Reply:
I forgot why exactly, but we decided back then to only copy the minimal needed information